RecoTracker/MkFitCore/src/MatriplexPackers.h

```diff
@@ -0,0 +1,152 @@
+#ifndef RecoTracker_MkFitCore_src_MatriplexPackers_h
+#define RecoTracker_MkFitCore_src_MatriplexPackers_h
+
+#include "Matrix.h"
+
+#include "RecoTracker/MkFitCore/interface/Hit.h"
+#include "RecoTracker/MkFitCore/interface/Track.h"
+
+namespace mkfit {
+
+  //==============================================================================
+  // MatriplexPackerSlurpIn
+  //==============================================================================
+
+  template <typename D>
+  class MatriplexPackerSlurpIn {
+  protected:
+    alignas(64) int m_idx[NN];
+
+    const D* m_base;
+    int m_pos;
+
+  public:
+    MatriplexPackerSlurpIn(const D& base) : m_base(&base), m_pos(0) {}
+
+    void reset() { m_pos = 0; }
+
+    void addNullInput() { m_idx[m_pos++] = 0; }
+
+    void addInput(const D& item) {
+      // Could issue prefetch requests here.
+
+      m_idx[m_pos] = &item - m_base;
+
+      ++m_pos;
+    }
+
+    void addInputAt(int pos, const D& item) {
+      while (m_pos < pos) {
+        // We might not care about initialization / reset to 0.
+        // Or we could be building an additional mask (on top of N_proc).
+        m_idx[m_pos++] = 0;
+      }
+
+      addInput(item);
+    }
+
+    template <typename TM>
+    void pack(TM& mplex, int base_offset) {
+      assert(m_pos <= NN);
+
+      if (m_pos == 0) {
+        // throw? ... this is probably not expected to happen.
```

Comment:
Could you remove?

Reply:
Check moved to assert.
Addressed in Round 4 follow-up PR trackreco/cmssw#79 commit: https://github.com/trackreco/cmssw/pull/79/commits/d081786b240bad6e7d367004a8b05c8de61ad18b